### PR TITLE
[12.0][ADD] base_google_map

### PR DESCRIPTION
**Pull request by murtuzasaleh**
Base Google Map
==============

This module allows you to setup Google Maps configuration.

### CODE REVIEW

**Comment by murtuzasaleh** [comment]
@dreispt @pedrobaeza @wolfhall @max3903

File: base_google_map/models/res_config_settings.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    google_maps_theme = fields.Selection(
+        selection=[('default', 'Default'),
+                   ('aubergine', 'Aubergine'),
+                   ('night', 'Night'),
+                   ('dark', 'Dark'),
+                   ('retro', 'Retro'),
+                   ('silver', 'Silver')],
+        string='Map theme')
+    google_maps_places = fields.Boolean(string='Places', default=True)
+    google_maps_geometry = fields.Boolean(string='Geometry', default=True)
```
Comment (dreispt) [inline base_google_map/models/res_config_settings.py:115]: I expected to see here additional fields, proposing the User to install the other options: web_widget_google_marker_icon_picker and web_widget_google_address_autocomplete.

**Review by dreispt — CHANGES_REQUESTED** [review]

File: base_google_map/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+{
+    'name': 'Base Google Map',
+    'version': '12.0.1.0.0',
+    'author': 'Open Source Integrators, Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/geospatial',
+    'license': 'AGPL-3',
+    'category': 'Extra Tools',
+    'depends': [
+        'base_setup',
+        'base_geolocalize',
```
Comment (dreispt) [inline base_google_map/__manifest__.py:13]: Should we also depend on `web_view_google_map` and have it automatically installed.
The module split is for maintainability, and I think the map view should immediately on install.

**Comment by murtuzasaleh** [comment]
@dreispt Can you please review this PR.

File: base_google_map/views/res_config_settings.xml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                                </div>
+                            </div>
+                            <div class="col-xs-12 col-md-6 o_setting_box">
+                                <div class="o_setting_left_pane">
+                                    <field name="module_web_widget_google_address_autocomplete"/>
+                                </div>
+                                <div class="o_setting_right_pane">
+                                    <label for="module_web_widget_google_address_autocomplete" string="Allow Google Address Form Autocomplete"/>
+                                    <div class="text-muted">
+                                        Allow Google Address Form Autocomplete
```
Comment (dreispt) [inline base_google_map/views/res_config_settings.xml]:
```suggestion
                                        Allow Google Address Autocomplete on Website Form (requires Website)
```

**Review by dreispt — APPROVED** [review]
Looks great, one one small suggestion.
Let's roll this out.

**Comment by dreispt** [comment]
@murtuzasaleh Afterthough: what about the demo on Contacts? Will that be on a separate module? Can it be included in the PR?

**Comment by murtuzasaleh** [comment]
> @murtuzasaleh Afterthough: what about the demo on Contacts? Will that be on a separate module? Can it be included in the PR?

It is included in [web_view_google_map
](https://github.com/OCA/geospatial/pull/223/commits/0db4bbe83742b24e42d4e6821a43214ec12ecf2c#diff-8d64706209419a236de8b867ababcc49R96) module.

**Comment by dreispt** [comment]
This is blocking 3 other PRs, so we need to move ahead with it.
I'll go ahead with a merge if no one opposes to it.

**Pull request merged**